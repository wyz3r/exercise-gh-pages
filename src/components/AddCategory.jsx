import { useState } from "react";

export function AddCategory({ onNewCategory }) {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length < 1) return;
    // setCategories((categories) => {
    //   const categoryExist = categories.find((c) => c === inputValue);
    //   if (categoryExist === undefined) return [inputValue, ...categories];
    //   const newArray = categories.filter((c) => c !== inputValue);
    //   console.log(inputValue);
    //   return [inputValue, ...newArray];
    // });
    
    setInputValue("");
    onNewCategory(inputValue)
  };
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search Gif"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
}
