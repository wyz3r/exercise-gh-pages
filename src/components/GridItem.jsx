
import PropTypes from 'prop-types'

export function GridItem({title, url}) {
    
  return (
    <div data-testid="card" className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}


GridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}