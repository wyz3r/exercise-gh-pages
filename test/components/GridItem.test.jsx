import { render, screen } from "@testing-library/react"
import { GridItem } from "../../src/components/GridItem"

describe(' testing component <GridItem />', () => { 
        const title = 'una_imagen'
        const url = 'url_del_Gif'
    test('should match with the snapshot', () => { 
        
        const {container} =  render(<GridItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
     })

     test('should show the Image ith the URL and ALT correct', () => { 

        render(<GridItem title={title} url={url}/>)
        // screen.debug()
        const {alt, src} = screen.getByRole('img')
        expect(src).toContain(url)
        expect(alt).toContain(title)


      })
      
      test('should show the title into the component ', () => { 

        render(<GridItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy()

       })

       test('should validate render the class "card" ', () => { 
        render(<GridItem title={title} url={url}/>)
        const div = screen.getByTestId('card');
        expect(div.getAttribute('class')).toBe('card')
        })
 })