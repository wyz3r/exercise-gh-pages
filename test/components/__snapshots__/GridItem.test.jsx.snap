// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` testing component <GridItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
    data-testid="card"
  >
    <img
      alt="una_imagen"
      src="url_del_Gif"
    />
    <p>
      una_imagen
    </p>
  </div>
</div>
`;
